### PR TITLE
Update default value transfer method

### PR DESCRIPTION
**Pull request by NicholasDotSol**
`.call.value()` should not revert in case of failure, but there is also no directly enforced gas limit.
It is important to ensure against reentrance for all value-handling functions.

- `send` also does not revert in case of failure, and there is an enforced gas limit that should limit reentrancy vectors, but it is not entirely reentrancy-secure due to the proven non-constant nature of op-code gas costs. Therefore instead of relying on `send` to restrict reentrancy, we explicitly ensure our contract is non-reentrant and allow a user's fallback to perform calculations beyond `send`'s limit since there is arguable no reason to enforce the gas limit.

### CODE REVIEW

File: solidity/contracts/deposit/DepositFunding.sol [hunk 1]
```diff
@@ -97,13 +97,13 @@ library DepositFunding {
...
-        _d.depositOwner().send(_d.keepSetupFee);
-        _d.pushFundsToKeepGroup(_seized.sub(_d.keepSetupFee));
-
         _d.setFailedSetup();
         _d.logSetupFailed();
 
+        /* solium-disable-next-line security/no-send */
+        _d.depositOwner().call.value(_d.keepSetupFee)("");
+        _d.pushFundsToKeepGroup(_seized.sub(_d.keepSetupFee));
+
```
Comment (Shadowfiend) [inline solidity/contracts/deposit/DepositFunding.sol:106]: We may want to leave a note as to why we're doing the state change in `fundingTeardown` after we do the above transfers.

File: solidity/contracts/deposit/DepositLiquidation.sol [hunk 2]
```diff
@@ -74,7 +74,7 @@ library DepositLiquidation {
         if (_d.auctionTBTCAmount() == 0) {
             // we came from the redemption flow
             _d.setLiquidated();
-            _d.redeemerAddress.transfer(_seized);
+            _d.redeemerAddress.call.value(_seized)("");
```
Comment (Shadowfiend) [inline solidity/contracts/deposit/DepositLiquidation.sol:77]: Feels like we should do this at the end of the function, per checks-effects-interactions, no?

File: solidity/contracts/deposit/DepositLiquidation.sol [hunk 3]
```diff
@@ -180,7 +180,7 @@ library DepositLiquidation {
 
         // Distribute funds to auction buyer
         uint256 _valueToDistribute = _d.auctionValue();
-        msg.sender.transfer(_valueToDistribute);
+        msg.sender.call.value(_valueToDistribute)("");
```
Comment (Shadowfiend) [inline solidity/contracts/deposit/DepositLiquidation.sol:183]: As above re: checks-effects-interactions.

File: solidity/contracts/deposit/DepositLiquidation.sol (hunk 2 above)
Comment (NicholasDotSol) [reply solidity/contracts/deposit/DepositLiquidation.sol:77]: we `setLiquidated` before transferring, preventing reentrancy. 

>  at the end of the function

Assuming you mean above `setFraudLiquidationInProgress`. I'm fine with that, although the values are unrelated.

File: solidity/contracts/deposit/DepositFunding.sol (hunk 1 above)
Comment (NicholasDotSol) [reply solidity/contracts/deposit/DepositFunding.sol:106]: there's no reason here. We can do it after as long as we ensure state change comes first.

File: solidity/contracts/deposit/DepositLiquidation.sol (hunk 3 above)
Comment (NicholasDotSol) [reply solidity/contracts/deposit/DepositLiquidation.sol:183]: `_d.setLiquidated();` is at the top of the function, preventing reentrancy.

Not really much more to move up aside from the initiator check.

**Comment by NicholasDotSol** [comment]
Closed in favor of #570
moving to a pull model

**Pull request closed without merging**